### PR TITLE
Add option to automatically run the automapper

### PR DESCRIPTION
**Pull request by bojidar-bg**
Also, put automapping config selection seperatelly from the automapper button.
Also, add seed parameter to the automapper.

<sup>~~Marked as WIP since I would like to save the selected automapper config and seed (and whether it should run automatically) in the .map file, but I'm not sure how it would be best to handle that. Input welcome.~~</sup>

### CODE REVIEW

**Comment by bojidar-bg** [comment]
Merged #1329 into my branch in order to add the automapper configuration as an external itemtype.

**Comment by heinrich5991** [comment]
From the glimpse I took on the diff: Map items only support `int` variables, not `bool`s.

**Comment by bojidar-bg** [comment]
Rebased and fixed.

File: src/game/editor/auto_map.cpp [hunk 1]
```diff
@@ -7,6 +8,35 @@
 #include "auto_map.h"
 #include "editor.h"
 
+// Based on triple32inc from https://github.com/skeeto/hash-prospector/tree/79a6074062a84907df6e45b756134b74e2956760
```
Comment (heinrich5991) [inline src/game/editor/auto_map.cpp]: Under UNLICENSE, so this is fine, afaict. https://github.com/skeeto/hash-prospector/blob/79a6074062a84907df6e45b756134b74e2956760/UNLICENSE

**Comment by def-** [comment]
Anyone feeling up to review?

**Comment by def-** [comment]
Let's merge it and see how it works out I guess ;)

**Pull request merged**